lib/spack/spack/util/environment.py

```diff
@@ -83,6 +84,16 @@ def double_quote_escape(s):
     return '"' + s.replace('"', r"\"") + '"'
 
 
+def system_env_normalize(func):
+    @wraps(func)
+    def marshall_env_var_name(self, name: str, *args, **kwargs):
```

Comment:
"marshal" typically refers to serialization AFAIK, I think a more-appropriate name would be `case_insensitive_modification`. I also think it would be great to augment the docstring with details that you provided in https://github.com/spack/spack/pull/36813#discussion_r1164761647 (mainly that this is for windows because setting `Path` and `PATH` will conflict).

Reply:
I was not aware of that convention, thanks for pointing it out, as I've evidently misused it a few places in Spack. That name sounds fine. Updated and pushed.